### PR TITLE
[MODFIN-350] Release/unrelease with new batch API

### PR DESCRIPTION
**Pull request by damien-git**
## Purpose
[MODFIN-350](https://issues.folio.org/browse/MODFIN-350) - Use the batch API for the release/unrelease encumbrance endpoints

## Approach
See ticket.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/services/transactions/BatchTransactionService.java [hunk 1]
```diff
@@ -1,21 +1,73 @@
...
+  public Future<Void> unreleaseEncumbrance(String id, RequestContext requestContext) {
+    return baseTransactionService.retrieveTransactionById(id, requestContext)
+      .compose(transaction -> unreleaseEncumbrance(transaction, requestContext))
+      .onSuccess(v -> logger.info("Unrelease encumbrance successful, id={}", id))
+      .onFailure(t -> logger.error("Unrelease encumbrance failed, id={}", id, t));
+  }
+
+  private Future<Void> releaseEncumbrance(Transaction transaction, RequestContext requestContext) {
+    baseTransactionService.validateTransactionType(transaction, Transaction.TransactionType.ENCUMBRANCE);
+    if (transaction.getEncumbrance().getStatus() == Encumbrance.Status.RELEASED) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/services/transactions/BatchTransactionService.java:51]: let's improve logging here, it always good practice to add log in conditional statement

File: src/main/java/org/folio/services/transactions/BatchTransactionService.java [hunk 2]
```diff
@@ -1,21 +1,73 @@
...
+      return succeededFuture(null);
+    }
+    transaction.getEncumbrance().setStatus(Encumbrance.Status.RELEASED);
+    return updateTransaction(transaction, requestContext);
+  }
+
+  private Future<Void> unreleaseEncumbrance(Transaction transaction, RequestContext requestContext) {
+    baseTransactionService.validateTransactionType(transaction, Transaction.TransactionType.ENCUMBRANCE);
+    if (transaction.getEncumbrance().getStatus() != Encumbrance.Status.RELEASED) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/services/transactions/BatchTransactionService.java:60]: let's improve logging here, it always good practice to add log in conditional statement

**Review by SerhiiNosko — APPROVED** [review]

File: src/main/java/org/folio/services/transactions/BatchTransactionService.java (hunk 1 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/services/transactions/BatchTransactionService.java:51]: added in PR to improve logging: https://github.com/folio-org/mod-finance/pull/230/files#diff-1e6f1e1816a583a7201d5f8e02f6712ca4d378753352f23abb536348492e948cR98
so nevermind, resolving this comment

File: ramls/release-encumbrance.raml [hunk 3]
```diff
@@ -7,7 +7,7 @@ baseUri: https://github.com/folio-org/mod-finance
 
 documentation:
   - title: Release encumbrance API
-    content: This documents the API calls that release any remaining money encumbered back to the budget's available pool
+    content: This documents the API calls that release any remaining money encumbered back to the budget's available pool; this should only be used by the UI, and only to release a single encumbrance - batch-all-or-nothing should be used otherwise
```
Comment (imerabishvili) [inline ramls/release-encumbrance.raml:10]: Can we have a multiline `content` to avoid long lines?

Comment (damien-git) [reply ramls/release-encumbrance.raml]: Yes. And I used the occasion to use `<tt>` for the new endpoint.

**Review by imerabishvili — APPROVED** [review]

**Pull request merged**